app/main.py

```diff
@@ -1,3 +1,15 @@
 def cache(func):
-    # Write your code here
-    pass
+    cache_results = {}
+
+    def wrapped(*args, **kwargs):
+        function_args = tuple([*args])
+        if function_args not in cache_results:
+            res = func(*args)
+            cache_results[function_args] = res
+            print("Calculating new result")
+            return res
+        else:
+            print("Getting from cache")
+            return cache_results.get(function_args)
```

Comment:
At this moment you are sure that args are in the dictionary, so you can use the following way to get the result
```suggestion
            return cache_results[function_args]
```